x-pack/plugins/maps/public/embeddable/map_embeddable_factory.ts

```diff
@@ -50,6 +53,11 @@ export class MapEmbeddableFactory implements EmbeddableFactoryDefinition {
 
   create = async (input: MapEmbeddableInput, parent?: IContainer) => {
     const { MapEmbeddable } = await lazyLoadMapModules();
+    const usageCollection = getUsageCollection();
+    if (usageCollection) {
+      const appId = await lastValueFrom(getApplication().currentAppId$.pipe(first()));
```

Comment:
how about the below instead

```
    // currentAppId$ is a BehaviorSubject exposed as an observable so subscription gets last value upon subscribe 
    getApplication().currentAppId$.pipe(first()).subscribe((appId: string) => {
      usageCollection.reportUiCounter('map', 'loaded', `open_maps_vis_${appId}`);
    });
```

Reply:
Even though the promise implementation is used in other places in Kibana, I agree that safety is paramount. Your suggestion works well and the subscription completes itself, so I've implemented it in 39e7548.